chapter2/tests/t3/chapter2-t3-velero-eks-diff.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Task 3 - find namespaces without the backup
+set -eo pipefail
+
+
+readonly SAVED_NS_URL='https://gist.github.com/dmitry-mightydevops/016139747b6cefdc94160607f95ede74/raw/velero.yaml'
+readonly ALL_NS_URL='https://gist.githubusercontent.com/dmitry-mightydevops/297c4e235b61982f21a0bbbf7319ac24/raw/kubernetes-namespaces.txt'
+
+
+# Function for printing error messages and exiting with a specific exit code
+# $1 - The exit code to use when exiting the script
+# $2 - The error message to print
+err() {
+  local code="$1"
+  shift
+  echo "[[ERROR]: $(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+  exit "$code"
+}
+
+# File exist check
+if curl -s --head  --request GET "$ALL_NS_URL" | grep "404" > /dev/null; then
```

Comment:
1. Please again create a separate function to retrieve any file with `curl`, so it will take `url` param and print common error that file was not found by passed url, because now you again have repetitions with absolutely the same logic
2. Also if curl will return 403 code? It will still be error, but you code will not raise error here. So it is better to check that returned code is not 200 and then raise error

Le't's make a separate function `get_file` and it will try to get file with curl and return its content if it can, otherwise it will raise error (i.e. you will make all curl requests to get file in one this function)

Reply:
Good, all done!